engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala

```diff
@@ -721,32 +685,54 @@ object SuggestionsHandler {
 
   /** The suggestion updates state.
     *
-    * @param suggestionUpdatesQueue the queue containing update messages
+    * @param suggestionUpdatesQueue the queue containing suggestion update messages
+    * @param suggestionLoadingQueue the queue containing notifications about loaded suggestions
     * @param isSuggestionUpdatesRunning a flag for a running suggestion update action
     * @param isSuggestionLoadingRunning a flag for a running suggestion loading action
     * @param shouldStartBackgroundProcessing a flag for starting a background
     * processing action
     */
-  final case class State(
-    suggestionUpdatesQueue: mutable.Queue[
-      Api.SuggestionsDatabaseModuleUpdateNotification
-    ] = mutable.Queue.empty,
-    suggestionLoadingQueue: mutable.Queue[
+  final private class State(
+    val suggestionUpdatesQueue: mutable.Queue[Any] = mutable.Queue.empty,
+    val suggestionLoadingQueue: mutable.Queue[
       Api.SuggestionsDatabaseSuggestionsLoadedNotification
-    ]                                        = mutable.Queue.empty,
-    isSuggestionUpdatesRunning: Boolean      = false,
-    isSuggestionLoadingRunning: Boolean      = false,
-    shouldStartBackgroundProcessing: Boolean = true
-  )
+    ]                                            = mutable.Queue.empty,
+    var isSuggestionUpdatesRunning: Boolean      = false,
```

Comment:
Maybe it would rather make sense to make all of those fields private (e.g. via `_is...`) and provide proper accessors?